app/controllers/application_controller.rb

```diff
@@ -1426,6 +1426,9 @@ def get_view(db, options = {}, fetch_data = false)
     sortcol_sym = "#{sort_prefix}_sortcol".to_sym
     sortdir_sym = "#{sort_prefix}_sortdir".to_sym
 
+    # Change db_sym for proper setting of @gtl_type as Service Catalogs and Catalog Items use the same db
+    db_sym = :catalog if x_active_tree == :svccat_tree && %w(servicetemplate).include?(db_sym.to_s)
```

Comment:
Look at line 1391:

db_sym = (options[:gtl_dbname] || dbname).to_sym # Get db name as symbol
you should be able to pass gtl_dbname in options to override db_sym there.

(So that this very service-specific code can live in the service(?) controller, instead of in application controller.)

Any reason you can't do that?

Reply:
I was expecting similar comment on this because yes, it would be nicer to have this change implemented in catalog controller, not here, in app controller. App controller is more general but I tried to push there this specific thing, because it is simple, straight, clear and it works well comparing to what I was trying to add in catalog controller. I better pushed here what really works. What I forgot was to move this PR to WIP till I find nicer and working solution (we both know now that I probably was doing something wrong that such a simple thing as passing gtl_dbname in options did not work for me). Thank you!

Reply:
:+1: thanks, looks much better now :)